runtime/src/main/scala/akka/grpc/netty/NettyCircuitBreakerInterceptor.scala

```diff
@@ -0,0 +1,57 @@
+package akka.grpc.netty
+
+import akka.pattern.CircuitBreaker
+import io.grpc.CallOptions
+import io.grpc.Channel
+import io.grpc.ClientCall
+import io.grpc.ClientInterceptor
+import io.grpc.ClientInterceptors
+import io.grpc.ForwardingClientCallListener
+import io.grpc.Metadata
+import io.grpc.MethodDescriptor
+import io.grpc.Status
+
+/**
+ * A `io.grpc.ClientInterceptor` that prevents downstream overload wrapping the calls with a circuit breaker.
+ */
+class NettyCircuitBreakerInterceptor(breaker: CircuitBreaker) extends ClientInterceptor {
+
+  private class Listener[RespT](delegate: ClientCall.Listener[RespT])
+      extends ForwardingClientCallListener.SimpleForwardingClientCallListener[RespT](delegate) {
+    override def onClose(status: Status, trailers: Metadata): Unit = {
+      if (status.isOk) {
+        breaker.succeed()
+      } else {
+        breaker.fail()
+      }
+
+      super.onClose(status, trailers)
+    }
+  }
+
+  override def interceptCall[ReqT, RespT](
+      method: MethodDescriptor[ReqT, RespT],
+      callOptions: CallOptions,
+      next: Channel): ClientCall[ReqT, RespT] =
+    new ClientInterceptors.CheckedForwardingClientCall[ReqT, RespT](next.newCall(method, callOptions)) {
+
+      override def checkedStart(responseListener: ClientCall.Listener[RespT], headers: Metadata): Unit = {
+        delegate().start(new Listener(responseListener), headers)
+      }
+
+      override def isReady: Boolean = {
+        if (breaker.isClosed || breaker.isHalfOpen) {
+          super.isReady
+        } else {
+          throw new IllegalStateException("Circuit Breaker is open. Can't send messages.")
```

Comment:
use `CircuitBreakerOpenException`?

Reply:
Yeah, this is one of the areas where I struggled the most. I think the exception is swallowed and treated differently by the grpc-java internals. I couldn't get to the exception I was throwing from user code so I settled to using a vanilla Java API one. 
If it makes no difference we can switch to `CircuitBreakerOpenException`.